docs/data/date-pickers/calendar-systems/AdapterHijri.tsx

```diff
@@ -1,31 +1,54 @@
 import * as React from 'react';
+import { prefixer } from 'stylis';
+import rtlPlugin from 'stylis-plugin-rtl';
+import createCache from '@emotion/cache';
+import { CacheProvider } from '@emotion/react';
 import moment from 'moment-hijri';
 import { AdapterMomentHijri } from '@mui/x-date-pickers/AdapterMomentHijri';
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
-import useTheme from '@mui/system/useTheme';
-import { createTheme, ThemeProvider } from '@mui/material/styles';
+import { createTheme, ThemeProvider, useTheme } from '@mui/material/styles';
+
+// Create rtl cache
+const cacheRtl = createCache({
+  key: 'adapter-hijri-demo',
+  stylisPlugins: [prefixer, rtlPlugin],
+});
 
 export default function AdapterHijri() {
+  // Inherit the theme from the docs site (dark/light mode)
   const existingTheme = useTheme();
+
   const theme = React.useMemo(
-    () => createTheme({ direction: 'rtl' }, existingTheme),
+    () => createTheme(existingTheme, { direction: 'rtl' }),
```

Comment:
The order was wrong, `existingTheme` contains `direction"=ltr"` so it overrode `{ direction: 'rtl' }`

This, plus the missing stylis plugin meant that our UI was basically a mix of LTR and RTL on those demos.